crates/networking/rpc/engine/payload.rs

```diff
@@ -355,23 +355,45 @@ impl GetPayloadRequest {
 
     fn handle(&self, context: RpcApiContext) -> Result<Value, RpcErr> {
         info!("Requested payload with id: {:#018x}", self.payload_id);
-        let Some(mut payload) = context.storage.get_payload(self.payload_id)? else {
+
+        let payload = context.storage.get_payload(self.payload_id)?;
+
+        let Some((mut payload_block, block_value, blobs_bundle, closed)) = payload else {
             return Err(RpcErr::UnknownPayload(format!(
                 "Payload with id {:#018x} not found",
                 self.payload_id
             )));
         };
+
         // Check timestamp matches valid fork
         let chain_config = &context.storage.get_chain_config()?;
-        let current_fork = chain_config.get_fork(payload.header.timestamp);
+        let current_fork = chain_config.get_fork(payload_block.header.timestamp);
         info!("Current Fork: {:?}", current_fork);
         if current_fork != self.valid_fork() {
             return Err(RpcErr::UnsuportedFork(format!("{current_fork:?}")));
         }
 
-        let (blobs_bundle, block_value) = build_payload(&mut payload, &context.storage)
+        if closed {
```

Comment:
nit: would it be better to call this `completed`?

Reply:
Yeah! I thought also of `build_ended`, but thought it was too long, `completed` seems better!

Reply:
Done!